src/MagentoHackathon/Composer/Magento/Patcher/Bootstrap.php

```
@@ -24,86 +55,158 @@ private function getConfig()
-            }
             if ($isBootstrapPart) {
-                $mageBootstrapFile .= $row;
+                $mageFileBootstrapPart .= $row;
             } else {
-                $mageClassFile .= $row;
+                $mageFileClassDeclarationPart .= $row;
             }
             if (strpos($row, 'Varien_Autoload') === 0) {
                 $isBootstrapPart = false;
             }
         }
-        $mageClassFile .= PHP_EOL;
-        $mageBootstrapFile .= PHP_EOL;
-        file_put_contents($appPath . '/Mage.class.php', $mageClassFile);
-        file_put_contents($appPath . '/Mage.bootstrap.php', $mageBootstrapFile);
 
-        $mageFileReplacement
-            = <<<php
-<?php
-require __DIR__ . '/bootstrap.php';
+        $mageFileReplacement = $mageFileBootstrapPart . PHP_EOL
+                             . $this->getAutoloaderPatchString() . PHP_EOL
+                             . $mageFileClassDeclarationPart;
 
-php;
-        file_put_contents($appPath . '/Mage.php', $mageFileReplacement);
+        return file_put_contents($this->getMageClassFilePath(), $mageFileReplacement) !== false;
+    }
 
+    /**
+     * @param IOInterface $io
+     */
+    public function setIo(IOInterface $io)
+    {
+        $this->io = $io;
+    }
 
+    /**
+     * @return IOInterface
+     */
+    public function getIo()
+    {
+        if (!$this->io) {
+            $this->io = new NullIO();
+        }
+        return $this->io;
     }
 
-    protected function generateBootstrapFile()
+    /**
+     * @return string
+     */
+    private function getAutoloaderPatchString()
     {
-        $appPath = $this->getAppPath();
-        $bootstrapFile
-            = <<<php
-<?php
-require __DIR__ . '/Mage.class.php';
-require __DIR__ . '/Mage.bootstrap.php';
+        $patchMark = self::PATCH_MARK;
+        return <<<PATCH
+/** $patchMark **/
+\$autoloaderPath = '{$this->getVendorAutoloaderPath()}';
+if (file_exists(\$autoloaderPath)) {
+    require_once \$autoloaderPath;
+}
```

The `file_exists` is there as a runtime check, so that the **app** won't fail. But when the patched is applied, that's another thing - then we know for sure the file exists (otherwise Composer can't function).

Still, I agree. Let's throw an exception in there.